sktime/forecasting/tests/test_naive.py

```diff
@@ -109,3 +109,29 @@ def test_strategy_mean_seasonal_simple(n_seasons, sp):
     y_pred = f.predict(fh)
 
     np.testing.assert_array_equal(y_pred, expected)
+
+
+@pytest.mark.parametrize("fh", TEST_OOS_FHS)
+@pytest.mark.parametrize("window_length",
+                         [*TEST_WINDOW_LENGTHS, None])
+def test_strategy_drift(fh, window_length):
+    if window_length != 1:
+        f = NaiveForecaster(strategy="drift",
+                            window_length=window_length)
+        f.fit(y_train)
+        y_pred = f.predict(fh)
+
+        if window_length is None:
+            window_length = len(y_train)
+
+        window_length = y_train.iloc[-window_length:]
```

Comment:
It's easy to write unit tests for the drift method without using the same code that we use inside the drift method. For example, we know the solution for a number of easy cases and can check if the forecaster returns the expected solution: 

1) a flat time series with zero slope, in this case strategy="drift" and "last" should be the same
2) a time series with constant slope 1, e.g. `np.arange(n_timepoints)` or `-np.arange(n_timepoints)`

If you want to support missing values too, we should add some tests for that as well.